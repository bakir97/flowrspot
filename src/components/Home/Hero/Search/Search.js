import React, { Component } from "react";
import styles from "./Search.module.scss";
import searchIcon from "../../../../assets/images/pl-icon-search.svg";
export default class Search extends Component {
  state = {
    searchTerm: ""
  };
  onSearcChange = e => {
    this.setState({ searchTerm: e.target.value });
  };
  submitSearch = e => {
    e.preventDefault();
    this.props.handleSearch(this.state.searchTerm);
  };
  render() {
    return (
      <form onSubmit={this.submitSearch} className={styles.container}>
        <input
          onChange={this.onSearcChange}
          className={styles.input}
          placeholder="Looking for something specific?"
        />
        <div className={styles.image_container}>
          <img
            src={searchIcon}
            className={styles.image}
            onClick={this.submitSearch}
            alt="search"
          />
        </div>
      </form>
    );
  }
}
