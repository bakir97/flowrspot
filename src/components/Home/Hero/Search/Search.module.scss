@import "../../../../assets/global/styles/mixins.scss";
.container {
  width: 600px;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 3px;
  box-shadow: 0 15px 30px 0 rgba(0, 0, 0, 0.05);
  background-color: white;
  @media screen and (max-width: 600px) {
    width: 307px;
  }
}
.input {
  @include input-style;
  margin-left: 19px;
  flex: 1;
  height: 60px;
  border: none;
  @include placeholder-styles;
  &:focus {
    outline: none;
  }
}
.image_container {
  margin-right: 19px;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}
.image {
  width: 20px;
  height: 20px;
  object-fit: contain;
  &:hover {
    cursor: pointer;
  }
}
