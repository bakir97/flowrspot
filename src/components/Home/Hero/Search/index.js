import Search from "./Search";
import { connect } from "react-redux";
import { searchFlowers } from "../../../../redux/flowers/actions";
const mapStateToProps = state => ({});

const mapDispatchToProps = {
  searchFlowers
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Search);
