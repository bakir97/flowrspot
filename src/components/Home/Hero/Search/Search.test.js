import React from "react";
import { shallow } from "enzyme";
import Search from "./Search";
describe("Search", () => {
  it("should render correctly", () => {
    const component = shallow(<Search />);
    expect(component).toMatchSnapshot();
  });
  it("should render form", () => {
    const component = shallow(<Search />);
    expect(component.exists("form")).toBe(true);
  });
  it("should change state on search", () => {
    const component = shallow(<Search />);
    component.instance().onSearcChange({ target: { value: "test" } });
    expect(component.state().searchTerm).toBe("test");
  });
});
