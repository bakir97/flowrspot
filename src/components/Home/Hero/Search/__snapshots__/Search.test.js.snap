// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly 1`] = `
<form
  className="container"
  onSubmit={[Function]}
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Looking for something specific?"
  />
  <div
    className="image_container"
  >
    <img
      alt="search"
      className="image"
      onClick={[Function]}
      src="pl-icon-search.svg"
    />
  </div>
</form>
`;
